### ISSUE TITLE
Mailer or HTTP sender not working when toast is used on android 10

### ISSUE THREAD

**Issue by ElavanResu** [ACRA/acra#801]
Hello,

I am using ACRA 5.1.3 and I have configured the files to use toast message and HTTP or Mailer sender. I am using Android 10 (Xiaomi POCO F1). Now whenever my app crashes it shows the toast message but it doesn't send any crash reports via HTTP sender or the Mail Sender (I know mail sender only works on Xiaomi when pop up permissions are enabled, I have also enabled that while testing).

But when I am using Dialog instead of Toast, it's sending reports via HTTP and Mailer. I don't understand why it doesn't send the report when toast is used.

I have also tested the toast configuration with mailer and HTTP sender on android 5 (Sony Xperia). It works perfectly on that phone. No issues with toast.

So, now I am confused whether it's the problem of toast with Xiaomi phones or the android 10 version.

**Comment by F43nd1r** [comment]
Use the newest version.

**Comment by ElavanResu** [comment]
Ahh well, bad news. I upgraded ACRA to 5.7.0. And tested the HTTP report sender with Toast configuration. But it didn't work in Xiaomi POCO F1 with android 10.

I did find something in logcat

`08-03 16:23:22.404  2035  2787 I AutoStartManagerService: MIUILOG- Reject service :Intent { cmp=com.coradius.agentapp/org.acra.sender.JobSenderService } userId : 0 uid : 10898
08-03 16:23:22.404  2035  2787 I JobServiceContext: MIUILOG- Reject Start Job, cancelJob uid : 10898 jobId :0
08-03 16:23:22.405  2035  2787 D JobScheduler: Error executing JobStatus{b5015eb #u0a898/0 com.coradius.agentapp/org.acra.sender.JobSenderService u=0 s=10898 TIME=none:-1ms READY}
08-03 16:23:22.406 26704 26733 I Process : Sending signal. PID: 26704 SIG: 9
08-03 16:23:22.406  2035  2787 D ActivityManager: report kill process: killerPid is:26704, killedPid is:26704
`

For some reason JobSenderService is getting rejected.

I arranged one samsung mobile with Android 10 and tested my ACRA configuration. It worked in that perfectly.

In conclusion, this problem only occur in Xiaomi mobile with Android 10.

**Comment by ElavanResu** [comment]
So, I'll list out the configuration working and not working on Xiaomi mobile android 10. Maybe this will help for someone else who's using Xiaomi android 10

Working:
1) With dialogue configuration, both HTTP and Mail sender works (For mail sender app needs permission "Display popup window while running in background")

2) With Silent configuration, both HTTP and Mail sender works perfectly. Mailer doesn't even need the permission that 1st point is needed.

Not working:
1) With toast configuration, only mail sender works. That too requires the permission "Display popup window while running in background"

**Comment by mikehardy** [comment]
Great research @ElavanResu - I am following this with interest. Xiaomi is so frustrating to me as a developer as they implement things differently (https://dontkillmyapp.com/xiaomi) and yet represent a fraction of the userbase large enough to special case. We have a mode in our app where HTTP sending is done with a toast, so your research will impact my project - thanks for reporting it

**Comment by mikehardy** [comment]
We ran a test on the issue you see linked from the Anki-Android repo above, and could not confirm this behaviour, at least with a 'MI 8' device (confirmed as Android 10 though). So I can't reproduce currently and have closed our related issue but hopefully as an extra data point this helps your investigation @ElavanResu  ?

**Comment by F43nd1r** [comment]
Superseded by #804
